src/contacts/tests/test_contacts.py

```diff
@@ -0,0 +1,333 @@
+        self.assertEqual(response.data, version.encode())
+
+    def test_ready_endpoint_200_status_code_ok_string(self):
+        """test if correct response is returned from readiness probe"""
+        response = self.test_app.get("/ready")
+        # assert 200 response code
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, b"ok")
+
+    def test_create_contact_201_status_code_correct_db_contact_object(self):
+        """test adding a new contact to a users contact list"""
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # send request to test client
+        response = self.test_app.post(
+            "/contacts/{}".format(EXAMPLE_USER),
+            headers=EXAMPLE_HEADERS,
+            data=json.dumps(example_contact),
+        )
+        # assert 201 response code
+        self.assertEqual(response.status_code, 201)
+        # assert contact object added to database had the required fields
+        # get the arg that contact_db.add_contact was called with
+        contact_object = self.mocked_db.return_value.add_contact.call_args[0][0]
+        # add username to example contact object
+        example_contact["username"] = EXAMPLE_USER
+        # assert all keys are equal
+        self.assertEqual(contact_object, example_contact)
+
+    def test_create_contact_409_status_code_add_same_user_to_contacts(self):
+        """test adding a contact with same account_num and routing_num as the user"""
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # set account_num of contact equal to account_num of user
+        example_contact["account_num"] = EXAMPLE_USER_PAYLOAD["acct"]
+        # set local routing number in service to match user routing number
+        self.flask_app.config["LOCAL_ROUTING"] = example_contact["routing_num"]
+        # send request to test client
+        response = self.test_app.post(
+            "/contacts/{}".format(EXAMPLE_USER),
+            headers=EXAMPLE_HEADERS,
+            data=json.dumps(example_contact),
+        )
+        # assert 409 response code
+        self.assertEqual(response.status_code, 409)
+        # assert we get correct error message
+        self.assertEqual(
+            response.json["msg"], "may not add yourself to contacts"
+        )
+
+    def test_create_contact_409_status_code_duplicate_contact_with_diff_label(self,):
+        """test adding a duplicate contact with same account_num
+            and routing_num but different label"""
+        # mock return value of get_contacts to return EXAMPLE_CONTACT
+        self.mocked_db.return_value.get_contacts.return_value = [
+            EXAMPLE_CONTACT
+        ]
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # change label of EXAMPLE_CONTACT while keeping other values
+        example_contact["label"] = "newlabel"
```

Comment:
a helper factory method with default arguments could be helpful here, along the lines of:
```
invalid_contact = newContact(label="new_label")
```

Reply:
So 9/15 tests use `example_contact = EXAMPLE_CONTACT.copy()` and label assignment is only done in 2/15 tests. I was thinking of putting `example_contact = EXAMPLE_CONTACT.copy()` into setup and letting label assignment as is as this will make it more concise than having a helper? wdyt?

Reply:
Also after reading your other comments, this above approach will not allow us to have different name for the `example_contact` like `invalid_contact` or `duplicate_contact`. So that is a drawback.

Reply:
Done, ended up going the helper route for more readability :)